enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,886 @@
+  no expectation that a cluster configured to use fewer than the
+  default resources would offer exactly the same performance as the
+  default. It must be stable and continue to operate reliably, but may
+  respond more slowly.
+* This enhancement does not address scaling the control plane **up**.
+
+## Proposal
+
+There are two aspects to manage scaling the control plane resource
+requests. Most of the cluster operators are deployed by the
+cluster-version-operator using static manifests that include the
+resource requests. Those operators then deploy other controllers or
+workloads dynamically. We need to account for both sets of components.
+
+### Deployments managed by the cluster-version-operator or operator-lifecycle-manager
+
+To start, we will [reduce the default
+requests](https://bugzilla.redhat.com/show_bug.cgi?id=1920159) for
+cluster operators deployed by the cluster-version-operator with static
+manifests. By re-tuning those settings by hand, we may be able to
+avoid implementing a more complex system to change them dynamically
+within the cluster.
+
+A similar approach can be taken for operators installed via the
+operator-lifecycle-manager.
+
+### Deployments managed by cluster operators
+
+We are considering 2 approaches for operands of cluster operators.
+
+#### Kubelet over-subscription configuration option
+
+Today, kubelet subtracts reserved CPUs from the available resources
+that it reports, under the assumption that they will be entirely
+consumed by workloads. Sometimes workloads do not entirely consume
+those CPUs, however, and non-guaranteed pods can also use the reserved
+CPU pool. A change to kubelet to provide a static over-subscription
+configuration option would provide a global approach to allowing the
+CPU resources reported by kubelet to be adjusted by a factor defined
+by the user to account for this difference, based on the workloads
+planned for the cluster.
+
+The new configuration option needs to be added to kubelet upstream in
+order to be supported long-term in OpenShift. A KEP will be written
+and submitted independently of this enhancement document.
+
+In OpenShift, the option would be set during installation and would be
+added to the control plane `MachineConfig`.
+
+We would need to carefully consider how to document the use of this
+option. Users will need to understand the minimum CPU resource
+requests of a base OpenShift deployment and any additional add-on
+operators before they will be able to tune the over-subscription
+setting so that all the components will fit onto the number of CPUs
+they require. We will need to keep the documentation of minimum CPU
+resource requests for OpenShift up to date for each release, or
+provide a tool to calculate the value from a default cluster.
+
+Refer to the implementation history section below for details of a
+proof-of-concept implementation of this approach.
+
+#### Cluster-wide resource request limit API
```

Comment:
Today we discussed a variation of this approach in which we do not add a new API but use the topology APIs to have operators reconfigure their operands to alternative static values. We would likely start with the top consumers and not try to update all operators in one release. This approach gives us more time to consider what a better scaling API should look like.